src/main/jit/Optimization.cpp

```diff
@@ -53,8 +85,8 @@ char BasicFunctionPass::pass_id = 0;
 //------------------------------------------------------------------------------
 // Impelementation of RemoveRedundantCallsToSetVisibility.
 
-bool RemoveRedundantCallsToSetVisibility::runOnBasicBlock(
-    llvm::BasicBlock& block) {
+int RemoveRedundantCallsToSetVisibility::RemoveRedundantCalls(
+    llvm::BasicBlock& block, bool leave_one) {
```

Comment:
In the (probably) common case where leave_one is false, this could be:

'''
for (llvm::Instruction& instr : block) {
  if (isCallToSetVisility(instr)) {
    instr.eraseFromParent();       
  }
}'''